src/SearchHistoryModal.module.css

```diff
@@ -0,0 +1,176 @@
+.container {
+  position: relative;
+}
+
+.iconButton {
+  background: none;
+  border: none;
+  cursor: pointer;
+  padding: var(--spacing-sm);
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border-radius: var(--borderRadius-sm);
+  transition: color 0.2s ease;
+}
+
+.modalOverlay {
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  width: 100vw;
+  height: 100vh;
+  background-color: var(--colors-bgSecondary);
+  display: flex;
+  justify-content: center;
+  align-items: flex-start;
+  z-index: 9999;
```

Comment:
I can't remember what we set for zindex stuff in the other parts of the app, but it would be better to put them in a base css file or something so they are all together and easier to see instead of in the individual module files in case you need to make something else the highest element on the page.